datafusion/core/src/catalog/listing_schema.rs

```diff
@@ -100,13 +102,20 @@ impl ListingSchemaProvider {
                 .ok_or_else(|| {
                     DataFusionError::Internal("Cannot parse file name!".to_string())
                 })?;
-            let table_name = table.to_str().ok_or_else(|| {
+            let table_name = file_name.split('.').collect_vec()[0];
```

Comment:
Deltatables are always folders, but `.csv`s should have their extension removed from their table name (and unfortunately the `Path` method to do this is marked unstable).

Reply:
Which method? Maybe we can contribute something back upstream to object_store 🤔

Reply:
https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.file_prefix

For some reason the github UI didn't let me respond in thread until now :/